### ISSUE TITLE
Using model.set_weights() to continue training at a later time

### ISSUE THREAD

**Issue by momih** [keras-team/keras#8575]
I'm using this [DenseNet model](https://github.com/flyyufelix/cnn_finetune/blob/master/densenet121.py). Due to issues with h5py, I am unable to use `model.save` to save my model after training. 

However I still need to be able to stop training and resume after a while. So in a way to save the model without using h5py, I pickle the weights from the model and then when I need to continue training the model from the last checkpoint I load the weights and set them in the model. The training section of my code roughly looks like this  - 

```
with open('weights_ckpt1', 'rb') as f:
    weights = pickle.load(f)

model = densenet121(img_rows=224, img_cols=224, color_type=1, num_classes=2)
model.set_weights(weights)

model.fit_generator(train_gen.flow(X_train, Y_train, batch_size=batch_size),
                    steps_per_epoch=len(X_train) / batch_size,
                    epochs=epochs,
                    verbose=1,
                    validation_data=validation_gen.flow(X_valid, Y_valid, batch_size=batch_size),
                    validation_steps=len(X_valid) / batch_size,
                    callbacks=[reducelr, early_stop, csvlog])

new_weights = model.get_weights()
with open('weights_ckpt2', 'wb') as f:
    pickle.dump(new_weights, f)
```

So when I started training, I did it for 5 epochs and pickled the weights in `weights_ckpt1`.  The loss achieved at the end of the 5th epoch was 0.66.
Now when I wanted to resume training, I loaded weights from `weights_ckpt1` and used `model.set_weights` . Now before first step of training, the loss was 0.66. But as soon as the first step began the loss increased to 1.06 and then slowly went back down 0.66 after 2-3 epochs. Why is this increase happening?

**Comment by mdfwn** [comment]
Your optimizer has different internal states that can change during training. I don't see what optimizer and what configurations you use, but I see that you use "reducelr" callback which changes the learning rate over time during training. So when you call "get_weights()" after the training has finished, your learning rate is smaller than at the beginning of the training, i.e. when you call "set_weights()". The higher learning rate makes your optimizer jump away from a better solution for couple of epochs.

So you need to figure out the learning rate at the end of your training and use this learning rate to initialize your optimizer at the beginning of the next training.
